kong/conf_loader/init.lua

```diff
@@ -1571,6 +1594,41 @@ local function load(path, custom_conf, opts)
     conf.loaded_plugins = setmetatable(plugins, _nop_tostring_mt)
   end
 
+  ---------------------------------
+  -- Dereference process references
+  ---------------------------------
+
+  do
+    local vault = require "kong.pdk.vault".new()
+    --local refs = {}
+    for k, v in pairs(conf) do
+      if vault.is_reference(v) then
+        local deref, deref_err = vault.get(v)
+        if deref ~= nil and not deref_err then
+          conf[k] = deref
+          --refs[k] = v
+        end
+
+        local schema_v = CONF_INFERENCES[k]
+        if schema_v then
+          if schema_v.typ ~= type(deref) then
+            -- TODO: what to do here? Fail hard or error log?
+            -- return nil, fmt("failed to dereference <%s> for config option %s. k, v)
+            log.error(fmt("referenced value has a wrong type, expected '%s', got '%s'", schema_v.typ, type(deref)))
```

Comment:
If we have secrets "active" (perhaps with a feature flag in the config?) then I think we should fail hard and explicitly here. Otherwise we risk leaving Kong in an unstable / insecure state.

```suggestion
            return nil, fmt("failed to dereference <%s> for config option %s. k, v)
```